dali/python/nvidia/dali/plugin/mxnet.py

```diff
@@ -72,12 +72,13 @@ def feed_ndarray(dali_tensor, arr, cuda_stream=None):
     ptr = ctypes.c_void_p()
     mx.base._LIB.MXNDArrayGetData(arr.handle, ctypes.byref(ptr))
 
+    non_blocking = cuda_stream is not None
     cuda_stream = types._raw_cuda_stream(cuda_stream)
 
     # Copy data from DALI tensor to ptr
     if isinstance(dali_tensor, (TensorGPU, TensorListGPU)):
         stream = None if cuda_stream is None else ctypes.c_void_p(cuda_stream)
-        dali_tensor.copy_to_external(ptr, stream, non_blocking=True)
+        dali_tensor.copy_to_external(ptr, stream, non_blocking=non_blocking)
```

Comment:
Is there a chance that `non_blocking` produces valid results when `stream` is None?